src/test/java/com/facebook/openwifirrm/modules/ModelerUtilsTest.java

```diff
@@ -77,4 +86,59 @@ void testValidMetric() throws Exception {
 		);
 		assertEquals(0.861, metric, 0.001);
 	}
+
+	@Test
+	void testBasicPreDot11nAggregatedWifiScanEntry() {
+		final long obsoletionPeriodMs = 900000;
+
+		final String apA = "aaaaaaaaaaaa";
+		final String bssidA = "aa:aa:aa:aa:aa:aa";
+		final String apB = "bbbbbbbbbbbb";
+		final String bssidB = "bb:bb:bb:bb:bb:bb";
+		DeviceDataManager deviceDataManager = new DeviceDataManager();
+		deviceDataManager.setTopology(TestUtils.createTopology("test-zone", apA, apB));
+		DataModel dataModel = new DataModel();
+
+		// if there are no scans (no list or empty list), there should be no aggregates
+		dataModel.latestWifiScans.computeIfAbsent(bssidB, k -> new LinkedList<>());
+		assertTrue(ModelerUtils.getAggregatedWifiScans(dataModel, obsoletionPeriodMs, new MeanAggregator()).isEmpty());
+
+		// for an AP, if there is one scan, the aggregate should just be that scan
+		// for the other device, there should be no aggregate
+		WifiScanEntry entryB1 = TestUtils.createWifiScanEntryWithBssid(1, bssidB); // defaults to signal = -60
+		dataModel.latestWifiScans.get(bssidB).add(Arrays.asList(entryB1));
+		WifiScanEntry aggregatedEntryB = ModelerUtils
+				.getAggregatedWifiScans(dataModel, obsoletionPeriodMs, new MeanAggregator()).get(bssidB);
+		assertFalse(dataModel.latestWifiScans.containsKey(bssidA));
+		assertEquals(entryB1, aggregatedEntryB);
+
+		// add another entry for device B and check the aggregation
+		WifiScanEntry entryB2 = TestUtils.createWifiScanEntryWithBssid(1, bssidB);
+		entryB2.signal = -62;
+		entryB2.unixTimeMs += 60000; // 1 min later
+		dataModel.latestWifiScans.get(bssidB).add(Arrays.asList(entryB2));
+		aggregatedEntryB = ModelerUtils.getAggregatedWifiScans(dataModel, obsoletionPeriodMs, new MeanAggregator())
+				.get(bssidB);
+		WifiScanEntry expectedAggregatedEntry = new WifiScanEntry(entryB2);
+		expectedAggregatedEntry.signal = -61; // average of -60 and -62
+		assertFalse(dataModel.latestWifiScans.containsKey(bssidA));
+		assertEquals(expectedAggregatedEntry, aggregatedEntryB);
+
+		// test the obsoletion period
+		WifiScanEntry entryB3 = TestUtils.createWifiScanEntryWithBssid(1, bssidB);
+		entryB3.signal = -64;
+		entryB3.unixTimeMs += obsoletionPeriodMs;
+		dataModel.latestWifiScans.get(bssidB).add(Arrays.asList(entryB3));
+		aggregatedEntryB = ModelerUtils.getAggregatedWifiScans(dataModel, obsoletionPeriodMs, new MeanAggregator())
+				.get(bssidB);
+		expectedAggregatedEntry = new WifiScanEntry(entryB3);
+		expectedAggregatedEntry.signal = -63; // average of -62 and -64
+		assertEquals(expectedAggregatedEntry, aggregatedEntryB);
+	}
+
+//	@Test
+//	void testMultipleEntriesAndAggregationLogic() {
+//		// TODO test 802.11n onwards (with non-null ht_oper and vht_oper)
```

Comment:
I will work on this for the next push (I just wanted to get a rough draft out since the aggregation logic has changed from my earlier PR)